src/pages/api/aws/sign-s3.ts

```diff
@@ -20,28 +24,38 @@ const s3 = new AWS.S3(options)
 
 const signedUrl = async (req: NextApiRequest, res: NextApiResponse) => {
   if (req.method === 'GET') {
-    const {objectName, contentType} = req.query
+    const user = await loadCurrentUser(req.cookies[ACCESS_TOKEN_KEY])
 
-    const filename = `${uuidv4()}/${objectName}`
+    const roles = intersection(user.roles, allRoles)
```

Comment:
when we load the roles from the server there are some exrtras so we do an "intersection" with the ones we are using here in this app to remove those extras and not throw an error